src/main/java/com/example/demo/repository/ShoppingCartRepository.java

```diff
@@ -0,0 +1,14 @@
+package com.example.demo.repository;
+
+import com.example.demo.model.ShoppingCart;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface ShoppingCartRepository extends JpaRepository<ShoppingCart, Long> {
+    @Query("SELECT s FROM ShoppingCart s LEFT JOIN s.cartItems c WHERE c.id = :userId")
```

Comment:
Have you checked how it works? I think this part is not correct `WHERE c.id = :userId`, because CartItem id can't be equal to User id

We need to use key-word FETCH, otherwise Hibernate will peform query with with JOIN, but won't load CartItem data and we still will have lazy loading (N + 1 problem) or even a LazyLoading exception. I also noticed, that in your `addItemToCart()` method you need Book data, so we need 2 joins here
```suggestion
    @Query("SELECT s FROM ShoppingCart s LEFT JOIN FETCH s.cartItems c WHERE s.id = :userId")
```